### ISSUE TITLE
Handle filenames containing backslashes

### ISSUE THREAD

**Issue by jpdillingham** [slskd/slskd#628]
Filenames containing backslashes will become a problem once #627 is merged, since backslashes will be used as the exclusive path separator character.

I believe the best approach is to, when `Path.DirectorySeparatorCharacter` is `/`, replace any backslashes with an underscore, e.g:

`/home/foo/Music/bar\baz.mp3`

Becomes:

`/home/foo/Music/bar_baz.mp3`

This value will be stored in the share cache and sent to peers over the network.  When a peer tires to download this file, the application should look first for `bar_baz.mp3`, and if it isn't found, look for `bar\baz.mp3`.

This will require refactoring the `ShareService.ResolveFilenameAsync()` method to check for the file on disk instead of just resolving the name from the cache.  If the file is not found before and after replacing underscores with backslashes, `NotFoundException` should be thrown.

After this is implemented, having files named `bar\baz.mp3` and `bar_baz.mp3` in the same directory is not allowed.

**Comment by jpdillingham** [comment]
Related Nicotine+ issue for the same problem: https://github.com/nicotine-plus/nicotine-plus/issues/1380

**Comment by jpdillingham** [comment]
Also consider url-encoding backslashes instead of replacing them with underscores.  `bar\baz.mp3` would become `bar%5Cbaz.mp3`.  This greatly reduces the risk of collision but makes the filename really hard to read and probably impossible to match programmatically.

**Comment by slook** [comment]
What happens when backslashes are in a Linux folder name?

**Comment by jpdillingham** [comment]
Great question! In a similar vein, what happens if more than one backslash is in the path?

I fiddled with this for a while yesterday and had a working solution where I stored the original filename as well as the filename with all backslashes replaced, and then did a lookup.  

This works in theory, but introduces a lot of extra processing when creating a browse response and significantly increases the amount of memory needed to store shares, which is already a big shortcoming of this app.

I'll make another attempt once my share cache is saved on disk.  For now I skip files and folders containing backslashes and log a warning about them not being supported.
